Realm/RLMResults.mm

```diff
@@ -563,6 +571,140 @@ - (RLMNotificationToken *)addNotificationBlock:(void (^)(RLMResults *, RLMCollec
     return _results.add_notification_callback(RLMWrapCollectionChangeCallback(block, self, true), std::move(keyPaths));
 }
 
+- (void)completionWithThreadSafeReference:(RLMThreadSafeReference * _Nullable)reference
+                               confinedTo:(RLMScheduler *)confinement
+                               completion:(RLMResultsCompletionBlock)completion
+                                    error:(NSError *_Nullable)error {
+    auto tsr = (error != nil) ? nil : reference;
+    RLMRealmConfiguration *configuration = _realm.configuration;
+    if (reference) {
+        RLMRealm *realm = [RLMRealm realmWithConfiguration:configuration confinedTo:confinement error:nil];
+        RLMResults *collection = [realm resolveThreadSafeReference:tsr];
+        collection.associatedSubscriptionId = self.associatedSubscriptionId;
+        [realm refresh];
+        [confinement invoke:^{
```

Comment:
Opening a Realm requires that we already be on the scheduler that we're opening the Realm for, so this can never be neccesary.